src/org.ts

```diff
@@ -3,38 +3,46 @@ import isValidDomain from 'is-valid-domain';
 import type { HydratedDocument } from 'mongoose';
 
 import { MemberAccessType, OrgModelSchema } from './models/Org.model.js';
-import type { OrgSchema } from './services/schema/org.schema.js';
+import type { OrgSchema, OrgSchemaPatch } from './services/schema/org.schema.js';
 import type { Result } from './utilities/result.js';
 import type { OrgCreationResult, ServiceOptions } from './orgTypes.js';
-import { CreationProblemType } from './CreationProblemType.js';
+import { OrgProblemType } from './OrgProblemType.js';
 
 const MONGODB_DUPLICATE_INDEX_CODE = 11_000;
 
 const MEMBER_ACCESS_TYPE_MAPPING: { [key in OrgSchema['memberAccessType']]: MemberAccessType } = {
   INVITE_ONLY: MemberAccessType.INVITE_ONLY,
   OPEN: MemberAccessType.OPEN,
+} as const;
+
+type ReversedMemberAccessType = {
+  [key in MemberAccessType]: OrgSchema['memberAccessType'];
+};
+
+const REVERSE_MEMBER_ACCESS_MAPPING: ReversedMemberAccessType = {
+  inviteOnly: 'INVITE_ONLY',
+  open: 'OPEN',
```

Comment:
we might write a better mapping in the mapper functions if we decide to go with them